### PR TITLE
Fixing deprecation and other warnings.

### PR DESCRIPTION
**Pull request by sduskis**

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java [hunk 1]
```diff
@@ -135,21 +135,6 @@ public static Credentials getCredentials(CredentialOptions options) throws IOExc
...
-   * Initializes OAuth2 credential using preconfigured ServiceAccount settings on the local
-   * Google Compute Engine VM. See:
-   * <a href="https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances"
-   * >Creating and Enabling Service Accounts for Instances</a>.
-   *
-   * @return a {@link com.google.auth.Credentials} object.
-   * @throws java.io.IOException if any.
-   * @throws java.security.GeneralSecurityException if any.
-   */
-  public static Credentials getCredentialFromMetadataServiceAccount()
```
Comment (kevinsi4508) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java]: Could this used publicly?

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java [hunk 2]
```diff
@@ -218,8 +208,7 @@ public static Credentials getApplicationDefaultCredential() throws IOException,
    * @throws java.io.IOException if any.
    * @throws java.security.GeneralSecurityException if any.
```
Comment (kevinsi4508) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java]: delete "@throws java.security.GeneralSecurityException if any."

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java (hunk 1 above)
Comment (sduskis) [reply bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java]: Users should not use CredentialFactory directly.  If they need to replicate this logic, they can use `ComputeEngineCredentials` themselves.  This code was a copy from somewhere else that needed it, but for CBT, we don't need this.

**Comment by sduskis** [comment]
Comments were addressed.  Thanks for the review.  PTAL.

**Comment by kevinsi4508** [comment]
LGTM

**Pull request merged**